subsys/net/l2/wifi/CMakeLists.txt

```diff
@@ -23,65 +23,61 @@ if (CONFIG_WIFI_NM)
-# Create output directory for test certs
-file(MAKE_DIRECTORY ${gen_dir})
+if(CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_ENTERPRISE)
+  # Wi-Fi Enterprise test certificates handling
+  set(gen_inc_dir ${ZEPHYR_BINARY_DIR}/misc/generated)
+  set(gen_dir ${gen_inc_dir}/wifi_enterprise_test_certs)
+  # Create output directory for test certs
+  file(MAKE_DIRECTORY ${gen_dir})
 
-# convert .pem files to array data at build time
-zephyr_include_directories(${gen_inc_dir})
+  # convert .pem files to array data at build time
+  zephyr_include_directories(${gen_inc_dir})
 
-if (CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_ENTERPRISE)
-generate_inc_file_for_target(
-    app
-    ${ZEPHYR_BASE}/samples/net/wifi/test_certs/client.pem
-    ${gen_dir}/client.pem.inc
-    )
+  generate_inc_file_for_target(
+      app
+      ${ZEPHYR_BASE}/samples/net/wifi/test_certs/client.pem
+      ${gen_dir}/client.pem.inc
+      )
 
-generate_inc_file_for_target(
-    app
-    ${ZEPHYR_BASE}/samples/net/wifi/test_certs/client-key.pem
-    ${gen_dir}/client-key.pem.inc
-    )
+  generate_inc_file_for_target(
+      app
+      ${ZEPHYR_BASE}/samples/net/wifi/test_certs/client-key.pem
+      ${gen_dir}/client-key.pem.inc
+      )
 
-generate_inc_file_for_target(
-    app
-    ${ZEPHYR_BASE}/samples/net/wifi/test_certs/ca.pem
-    ${gen_dir}/ca.pem.inc
-    )
+  generate_inc_file_for_target(
+      app
+      ${ZEPHYR_BASE}/samples/net/wifi/test_certs/ca.pem
+      ${gen_dir}/ca.pem.inc
+      )
 
-generate_inc_file_for_target(
-    app
-    ${ZEPHYR_BASE}/samples/net/wifi/test_certs/client2.pem
-    ${gen_dir}/client2.pem.inc
-    )
+  generate_inc_file_for_target(
+      app
+      ${ZEPHYR_BASE}/samples/net/wifi/test_certs/client2.pem
+      ${gen_dir}/client2.pem.inc
+      )
 
-generate_inc_file_for_target(
-    app
-    ${ZEPHYR_BASE}/samples/net/wifi/test_certs/client-key2.pem
-    ${gen_dir}/client-key2.pem.inc
-    )
+  generate_inc_file_for_target(
+      app
+      ${ZEPHYR_BASE}/samples/net/wifi/test_certs/client-key2.pem
+      ${gen_dir}/client-key2.pem.inc
+      )
 
-generate_inc_file_for_target(
-    app
-    ${ZEPHYR_BASE}/samples/net/wifi/test_certs/ca2.pem
-    ${gen_dir}/ca2.pem.inc
-    )
+  generate_inc_file_for_target(
+      app
+      ${ZEPHYR_BASE}/samples/net/wifi/test_certs/ca2.pem
+      ${gen_dir}/ca2.pem.inc
+      )
 
-generate_inc_file_for_target(
-    app
-    ${ZEPHYR_BASE}/samples/net/wifi/test_certs/server.pem
-    ${gen_dir}/server.pem.inc
-    )
+  generate_inc_file_for_target(
+      app
+      ${ZEPHYR_BASE}/samples/net/wifi/test_certs/server.pem
+      ${gen_dir}/server.pem.inc
+      )
 
-generate_inc_file_for_target(
-    app
-    ${ZEPHYR_BASE}/samples/net/wifi/test_certs/server-key.pem
-    ${gen_dir}/server-key.pem.inc
-    )
-# Add explicit dependency on app target for ZEPHYR_CURRENT_LIBRARY, so these
-# headers are generated at the correct point in the build
-add_dependencies(${ZEPHYR_CURRENT_LIBRARY} app)
```

Comment:
Why remove this `add_dependencies(${ZEPHYR_CURRENT_LIBRARY} app)`?
Which can fix the CI build error that xxx.pem.inc included but not generated yet.

Reply:
Removed by mistake. Added it back.